extensions-core/caffeine-cache/src/main/java/io/druid/client/cache/CacheExecutorFactory.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.client.cache;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import io.druid.concurrent.Execs;
+import java.util.concurrent.Executor;
+
+public enum CacheExecutorFactory
+{
+  COMMON_FJP {
+    @Override
+    public Executor createExecutor()
+    {
+      return null;
```

Comment:
I would return `ForkJoinPool.commonPool()` as `null` is subtle. It won't have any negative effect by specifying it explicitly in the builder. This would remove the if-else when constructing the cache.